commands/member/cabbages.js

```diff
@@ -1,13 +1,249 @@
-const { SlashCommandBuilder } = require('discord.js');
+const {
+    SlashCommandBuilder,
+    EmbedBuilder,
+    ActionRowBuilder,
+    ButtonBuilder,
+    ButtonStyle,
+    ComponentType,
+} = require('discord.js');
+const { getDiscordClient } = require('../../discord');
+const {
+    getCabbageBreakdown,
+    cabbagesUntilNext,
+} = require('../../helpers/calculateCabbages');
 const models = require('../../models');
 
+const findEmoji = (client, name) => {
+    const emoji = client.emojis.cache.find(
+        (emoji) => emoji.name.toLowerCase() === name.toLowerCase()
+    );
+    return emoji || '';
+};
+
+const capitalize = (string) => {
+    return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase();
+};
+
+const pad = (input, spaces) => {
+    // Make sure input is converted to a string
+    const string = typeof input === 'string' ? input : input.toString();
+    if (string.length == spaces) {
+        return string;
+    } else if (spaces - string.length == 1) {
+        return pad(string + ' ', spaces);
+    } else if (spaces - string.length >= 2) {
+        return pad(' ' + string + ' ', spaces);
+    } else {
+        // string.length > spaces
+        return ' '.repeat(spaces);
+    }
+};
```

Comment:
This might be cleaner rewritten as a switch statement. Thoughts?

Reply:
```js
const pad = (input, spaces) => {
    // Make sure input is converted to a string
    let string = typeof input === 'string' ? input : input.toString();
    if (string.length > spaces) return input.substring(1, spaces);
    // Add spaces until you need either 0 or 1 more
    while (string.length < spaces - 1) {
        string = ' ' + string + ' ';
    }
    if (string.length == spaces) return string;
    return string + ' ';
};
```
I had a look at the pad function and getting rid of the recursion is probably even better